lib/r10k/git/stateful_repository.rb

```diff
@@ -62,4 +61,13 @@ def status
       :insync
     end
   end
+
+  private
+
+  def sync?
+    return true if !@cache.exist?
+    return true if !(sha = @cache.resolve(@ref))
```

Comment:
I don't love inline assignment, but I see it's prevalent in the code. I'm not going to ask you to change it.